MSAL/src/public/MSALWebviewType.h

```diff
@@ -22,21 +20,31 @@
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
-//
-//------------------------------------------------------------------------------
-
-#import <Foundation/Foundation.h>
-
-typedef void (^MSALWebUICompletionBlock)(NSURL *response, NSError *error);
-
-@interface MSALWebUI : NSObject
 
-+ (void)startWebUIWithURL:(NSURL *)url
-                  context:(id<MSALRequestContext>)context
-          completionBlock:(MSALWebUICompletionBlock)completionBlock;
+#ifndef MSALWebviewType_h
+#define MSALWebviewType_h
 
-+ (BOOL)handleResponse:(NSURL *)url;
+typedef NS_ENUM(NSInteger, MSALWebviewType)
+{
+    // Use WKWebView
+    MSALWebviewTypeWKWebView,
+    
+#if TARGET_OS_IPHONE
+    
+    // Use SFAuthenticationSession/ASWebview
+    MSALWebviewTypeAuthenticationSession,
+    
+    // Use SFSafariViewController for all versions.
+    MSALWebviewTypeSafariViewController,
+    
+#endif
+    
+    // For iOS 11 and up, uses AuthenticationSession (ASAuthenticationSession or
+    // SFAuthenticationSession).
+    // For older versions, with AuthenticationSession not being available, uses
+    // SafariViewController.
+    MSALWebviewTypeAutomatic
```

Comment:
Should we call it "auto" or "automatic" :)?

Reply:
Automatic is fine for now I think